meeteval/viz/__main__.py

```diff
@@ -0,0 +1,211 @@
+from pathlib import Path
+import collections
+
+import meeteval
+from meeteval.viz.visualize import AlignmentVisualization
+from meeteval.wer.api import _load_texts
+import tqdm
+
+
+def create_viz_folder(
+        reference,
+        hypothesiss,
+        out,
+        alignment='tcp',
+        regex=None,
+):
+    out = Path(out)
+    out.mkdir(parents=True, exist_ok=True)
+
+    avs = {}
+    for i, hypothesis in tqdm.tqdm(hypothesiss.items()):
+
+        r, h = _load_texts(
+            reference, hypothesis, regex=regex,
+            reference_sort='segment',
+            hypothesis_sort='segment',
+        )
+
+        r = r.groupby('session_id')
+        h = h.groupby('session_id')
+
+        session_ids = set(r.keys()) & set(h.keys())
+        xor = set(r.keys()) ^ set(h.keys())
+        if xor:
+            print(f'Ignore {xor}, because they are not available in reference and hypothesis.')
+
+        for session_id in session_ids:
+            av = AlignmentVisualization(r[session_id],
+                                        h[session_id],
+                                        alignment=alignment)
+            av.dump(out / f'{session_id}_{i}.html')
+            avs.setdefault(i, {})[session_id] = av
+
+    # meeteval.wer.combine_error_rates([
+    #     meeteval.wer.ErrorRate.from_dict(av.data['info']['wer'] for av in avs.values())
+    #
+    # ])
+
+    ###########################################################################
+
+    from yattag import Doc
+    from yattag import indent
+
+    avs_T = collections.defaultdict(lambda: {k: None for k in avs.keys()})
+    for i, v in avs.items():
+        for session_id, av in v.items():
+            avs_T[session_id][i] = av
+    avs_T = dict(avs_T)
+
+    for session_id, v in avs_T.items():
+        doc, tag, text = Doc().tagtext()
+        doc.asis('<!DOCTYPE html>')
+        with tag('html'):
+            with tag('style'):
+                doc.asis("""html, body {height: 100%; margin: 0;}""")
+                doc.asis(
+                    """.full-height {height: 100%; background: yellow;}""")
+            with tag('body',
+                     style="width: 100%; height: 100%; display: flex;"):
+                for i, av in v.items():
+                    with tag('div',
+                             style=f"width: {100 // len(v)}%; height: 100%;"):
```

Comment:
You can use the flex layout to have a varying number of columns without explicit computations. Example:
```html
<html>
<style>
    .grid {
        height: 100%;
        display: flex;
    }
    .column {
        flex-grow: 1;
    }
    div {
        border: 1px solid black;
    }
</style>
<body>
<div class="grid">
    <div class="column">Column 1</div>
    <div class="column">Column 2</div>
    <div class="column">Column 3</div>
</div>
</body>
</html>
```

Reply:
Changed.